pkg/storage/etcd3/watcher.go

```diff
@@ -180,6 +180,9 @@ func (wc *watchChan) startWatching() {
 			wc.sendEvent(parseEvent(e))
 		}
 	}
+	wc.cancel()
```

Comment:
line 172-177 is the path with watch response; this line not.

We also don't simply call cancel() when receiving error. Because we need to send the final error via ResultChan() and then cancel().